go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -561,16 +562,17 @@ func runOvnKube(ctx context.Context, runMode *ovnkubeRunMode, ovnClientset *util
 		}()
 	}
 
+	ovsClient, err := ovs.GetOVSClient(ctx)
```

Comment:
Prefer if you pass this just like we do with everything else.

NewNodeControllerManager -> NewDefaultNodeNetworkController -> NewCNIServer. And form here you just pass ti around same as the clientset.

I don't have anything against singletons, but I would prefer not start popping them without some kind of central registry design or something. Otherwise its easy to lose track of them.

I think passing it around is the best approach for now.

Reply:
Also I don't think you need to change anything relative to `cancel()`. Just leave as it was. If anything, you can remove the call there was online 568 which should be completely unnecessary.